src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/CompareAttribute.cs

```diff
@@ -56,35 +56,23 @@ protected override ValidationResult IsValid(object value, ValidationContext vali
             {
                 if (OtherPropertyDisplayName == null)
                 {
-                    OtherPropertyDisplayName = GetDisplayNameForProperty(validationContext.ObjectType, OtherProperty);
+                    OtherPropertyDisplayName = GetDisplayNameForProperty(validationContext.ObjectType, otherPropertyInfo);
```

Comment:
If you look above, we have the following code:

```
var otherPropertyInfo = validationContext.ObjectType.GetRuntimeProperty(OtherProperty);
```

In the `GetDisplayNameForProperty` method, we used to have the buggy code:

```
var property = containerType.GetRuntimeProperties()
 .SingleOrDefault(
    prop =>
        IsPublic(prop) &&
        string.Equals(propertyName, prop.Name, StringComparison.OrdinalIgnoreCase));
```

However, we already have the property `otherPropertyInfo` above. The name comparison is broken, and `GetRuntimeProperty` only returns public methods so this whole thing is redundant and broken.

So we can just use the actual PropertyInfo we found before